### PR TITLE
front

### PR DESCRIPTION
**Pull request by svyatoslavvdovichenko**

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
```
Comment (baklanovairina) [inline index.html:4]: check tabs

File: script.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
+let activeEditTask = null;
+let allTasks = JSON.parse(localStorage.getItem('tasks')) || [];
+let input = null;
+let valueInput = '';
```
Comment (baklanovairina) [inline script.js:4]: why make these variables global? They are rarely used

File: script.js [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+
+
```
Comment (baklanovairina) [inline script.js:7]: remove extra spaces

File: script.js [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+window.onload = function  init () {
```
Comment (baklanovairina) [inline script.js:8]: use arrow function

File: script.js [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+  input = document.getElementById('add-task');
+  input.addEventListener('change', updateValue);
+  render();
+  
+  
+}
+
+onClickBtn = () => {
```
Comment (baklanovairina) [inline script.js:16]: where are const and let identifiers for functions?

onClickBtn - rename, it is not clear from the name what the function does

File: script.js [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+  allTasks.push({
+    text: valueInput,
+    isCheck: false,
+  });
+  localStorage.setItem('tasks', JSON.stringify(allTasks));
+  valueInput = '';
+  input.value = '';
```
Comment (baklanovairina) [inline script.js:23]: valueInput and input.value are the same. Both values are responsible for the input field

File: script.js [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+}
+
+updateValue = (event) => {
+    
+  valueInput = event.target.value;
+
+}
+
+render = () => {
+  const content = document.getElementById('content-page');
```
Comment (baklanovairina) [inline script.js:41]: getElementById may return null, so a null check is needed

File: script.js [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+  
+  while(content.firstChild) {
+    content.removeChild(content.firstChild);
+  }
+
+  allTasks.sort ((a, b) => a.isCheck > b.isCheck ? 1 : a.isCheck < b.isCheck ? -1 : 0)
```
Comment (baklanovairina) [inline script.js:47]: the sort function mutates the array, and mutation should be avoided. In this case, create a copy of the array and work with it. If there are mutations, it is difficult to find when searching for bugs where the array has changed

File: script.js [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+  allTasks.map((item, index) => {
```
Comment (baklanovairina) [inline script.js:48]: why map?

File: script.js [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+    const container = document.createElement('div');
+    container.id = `task = ${index}`;
```
Comment (baklanovairina) [inline script.js:50]: mistake

File: script.js [hunk 11]
```diff
@@ -0,0 +1,137 @@
...
+    const checkbox = document.createElement('input');
+
+    checkbox.type = 'checkbox';
+    checkbox.checked = item.isCheck;
+    checkbox.onchange = () => {
+    onChangeCheckbox(index);
+    
+    };
+    
+
```
Comment (baklanovairina) [inline script.js:62]: remove extra spaces

File: script.js [hunk 12]
```diff
@@ -0,0 +1,137 @@
...
+      container.appendChild(checkbox);
+      const text = document.createElement('p');
+      
+      text.innerText = item.text;
+      text.className = item.isCheck ? 'done-text' : 'text';
+      container.appendChild(text);
+      const imageDelete = document.createElement('img');
+      imageDelete.src = 'sourse/free-icon-close-4013407.png';
+      
+      imageDelete.onclick = () => {
```
Comment (baklanovairina) [inline script.js:80]: Pictures cannot be clickable. Create a button and add an icon to it. Button can be clickable

In addition, each image must have an alt attribute.

File: script.js [hunk 13]
```diff
@@ -0,0 +1,137 @@
...
+        onDeleteTask(index);
+      }
+      container.appendChild(imageDelete);
+
+    }
+
+    if (!item.isCheck) {
+      if (index === activeEditTask) {
```
Comment (baklanovairina) [inline script.js:88]: avoid such nesting, it worsens the readability of the code

File: script.js [hunk 14]
```diff
@@ -0,0 +1,137 @@
...
+          doneEditTask();
+        };
+        container.appendChild(imageDone);
+      } else {
+        const imageEdit = document.createElement('img');
+        imageEdit.src = 'sourse/pencil.png';
+        imageEdit.onclick = () => {
+          activeEditTask = index;
+          render();
+        };
```
Comment (baklanovairina) [inline script.js:101]: element editing should not be in render. This is a separate function.

File: style.css [hunk 15]
```diff
@@ -0,0 +1,31 @@
+body {
+    background-color: rgb(185,217,190);
```
Comment (baklanovairina) [inline style.css:2]: check tabs

File: style.css [hunk 16]
```diff
@@ -0,0 +1,31 @@
...
+.text-task {
+    font-size: 18px;
+    align-items: center;
+}
+
+.done-text {
+    color: rgba(0, 0, 0, 0.3);
+    text-decoration: line-through;
+}
+input {
```
Comment (baklanovairina) [inline style.css:23]: use BEM

**Review by svyatoslavvdovichenko — COMMENTED** [review]
/

**Comment by svyatoslavvdovichenko** [comment]
> /

**Pull request closed without merging**